pkg/controllers/placement/namespace_status_sync_test.go

```diff
@@ -42,328 +41,495 @@ var (
+				Client: fakeClient,
+				Scheme: scheme,
+			}
+
+			// Get the CRP from the fake client to ensure it has proper metadata and when handleNamespaceAccessibleCRP is called CRP must already exist.
+			gotCRP := &placementv1beta1.ClusterResourcePlacement{}
+			if err := fakeClient.Get(context.Background(), types.NamespacedName{Name: tc.placementObjName}, gotCRP); err != nil {
+				t.Fatalf("Failed to get CRP from fake client: %v", err)
+			}
+
+			// Call handleNamespaceAccessibleCRP with the CRP from the client.
+			err := reconciler.handleNamespaceAccessibleCRP(context.Background(), gotCRP)
+
+			// Check if error expectation matches.
+			if tc.wantError && err == nil {
+				t.Fatal("Expected error but got none")
+			}
+			if !tc.wantError && err != nil {
+				t.Fatalf("handleNamespaceAccessibleCRP() failed: %v", err)
+			}
+
+			// Check ClusterResourcePlacementStatus creation/update using cmp.Diff.
+			if tc.wantCRPSOperation && tc.targetNamespace != "" {
+				gotCRPS := &placementv1beta1.ClusterResourcePlacementStatus{}
+				getErr := fakeClient.Get(context.Background(), types.NamespacedName{
+					Name:      tc.placementObjName,
+					Namespace: tc.targetNamespace,
+				}, gotCRPS)
+
+				if getErr != nil {
+					if !tc.wantError {
+						t.Fatalf("Expected ClusterResourcePlacementStatus to be created/updated but got error: %v", getErr)
+					}
+				} else {
+					// Construct expected CRPS based on the CRP.
+					wantCRPS := &placementv1beta1.ClusterResourcePlacementStatus{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      tc.placementObjName,
+							Namespace: tc.targetNamespace,
+							OwnerReferences: []metav1.OwnerReference{
+								{
+									APIVersion:         "placement.kubernetes-fleet.io/v1beta1",
+									Kind:               "ClusterResourcePlacement",
+									Name:               tc.placementObjName,
+									Controller:         ptr.To(true),
+									BlockOwnerDeletion: ptr.To(true),
+								},
+							},
+						},
+						PlacementStatus: gotCRP.Status,
+					}
+
+					// Filter out StatusSynced condition from the expected CRPS.
+					filteredConditions := make([]metav1.Condition, 0)
+					for _, cond := range gotCRP.Status.Conditions {
+						if cond.Type != string(placementv1beta1.ClusterResourcePlacementStatusSyncedConditionType) {
+							filteredConditions = append(filteredConditions, cond)
+						}
+					}
+					wantCRPS.PlacementStatus.Conditions = filteredConditions
+
+					// Use cmp.Diff to compare CRPS objects.
+					if diff := cmp.Diff(wantCRPS, gotCRPS, crpsCmpOpts...); diff != "" {
+						t.Fatalf("ClusterResourcePlacementStatus mismatch (-want +got):\n%s", diff)
+					}
+
+					// Additional validation that LastUpdatedTime is set (ignored in comparison).
+					if gotCRPS.LastUpdatedTime.IsZero() {
+						t.Fatal("Expected LastUpdatedTime to be set on CRPS")
+					}
+				}
+			}
+
+			// Check StatusSynced condition on the original CRP using cmp.Diff.
+			if tc.wantCRPCondition != nil {
```

Comment:
I don't think we need this check? We just put the wanted condition in test case and compare with what we got? In this case, it works for nil condition too.